runtime/src/bank.rs

```diff
@@ -5315,7 +5290,6 @@ impl Bank {
             }
             rent_debits.insert(&pubkey, collected.rent_amount, account.lamports());
         }
-        metrics.hold_range_us.fetch_add(hold_range.as_us(), Relaxed);
         metrics.collect_us.fetch_add(collect_us, Relaxed);
         metrics.hash_us.fetch_add(hash_skipped_rewrites_us, Relaxed);
         metrics.store_us.fetch_add(store_us, Relaxed);
```

Comment:
(GitHub will not let me select and comment on the whole range that I want to. This should extend down to the end of the new `collect_rent_from_accounts()` fn.)

One improvement that can be made to this fn is to return all the metrics from each invocation of this function, instead of updating them here. Since they are all atomics, and being run in parallel, there likely will be a lot of cache evictions bouncing around at the end of the fn as each thread tries to update each atomic, which is likely non-trivial for more than a few threads. (Especially if these atomics live on the same cache lines.)

If this fn returns the metrics, then the parallel loop (`thread_pool.install()`) in `collect_rent_in_partition()` can collect them all up and perform a single round of atomic updates after the parallel loop is done.

Reply:
ok, try this